src/client/Microsoft.Identity.Client/Instance/Authority.cs

```diff
@@ -82,9 +83,15 @@ protected Authority(AuthorityInfo authorityInfo)
 
                 case AuthorityType.Aad:
 
+                    bool updateAuthority = requestContext.ServiceBundle.Config.MultiCloudSupportEnabled && account != null;
+
                     if (requestAuthorityInfo == null)
                     {
-                        return CreateAuthorityWithTenant(configAuthorityInfo, requestHomeAccountTenantId);
+                        return CreateAuthorityWithTenant(
+                            updateAuthority ? 
+                                new AuthorityInfo(AuthorityType.Aad, $"https://{account.Environment}/{account.HomeAccountId.TenantId}", configAuthorityInfo.ValidateAuthority) : 
```

Comment:
Can you move this logic in Authority or AuthorityInfo? We should not do any sort of string manipulation in here. There is already something very similar (and more robust) - https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/master/src/client/Microsoft.Identity.Client/Instance/Authority.cs#L165

Reply:
Updated